Frontend/student-hub/src/components/Signup.js

```diff
@@ -25,20 +29,23 @@ function Signup() {
 
       if (response.ok) {
         setOpen(true);
+        setLoading(false);
       } else {
         const { error } = await response.json();
         setError(error);
+        setLoading(false);
       }
     } catch (error) {
       console.error('Signup failed:', error);
       setError('Signup failed. Please try again.');
+      setLoading(false);
     }
```

Comment:
looks like we call `setLoading(false)` in every possible case. should we just put it once right after the fetch, so on L29?